codegen/cmd_test.go

```diff
@@ -41,6 +42,90 @@ var _ = Describe("Cmd", func() {
 	skv2Imports.External["github.com/solo-io/cue"] = []string{
 		"encoding/protobuf/cue/cue.proto",
 	}
+	It("with extra resources", func() {
+		cmd := &Command{
+			Chart: &Chart{
+				Data: Data{
+					ApiVersion:  "v1",
+					Description: "",
+					Name:        "Painting Operator",
+					Version:     "v0.0.1",
+					Home:        "https://docs.solo.io/skv2/latest",
+					Sources: []string{
+						"https://github.com/solo-io/skv2",
+					},
+				},
+				Operators: []Operator{{
+					Name: "painter",
+					Deployment: Deployment{
+						Container: Container{
+							Image: Image{Repository: "painter", Tag: "v0.0.1"},
+						},
+					},
+					ConditionalResources: []ConditionalResource{{
+						Condition: "$.Values.painter.enabled",
+						Object: &corev1.PersistentVolumeClaim{
+							TypeMeta: PersistentVolumeClaimType,
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "some-pvc",
+								Namespace: `{{ .Release.Namespace }}`,
+								Labels: map[string]string{
+									"app":       "gloo-mesh-redis",
+									"gloo-mesh": "gloo-mesh-redis",
+								},
+								Annotations: map[string]string{
+									"app.kubernetes.io/name": "gloo-mesh-redis",
+								},
+							},
+							Spec: corev1.PersistentVolumeClaimSpec{
+								AccessModes: []corev1.PersistentVolumeAccessMode{
+									corev1.ReadWriteOnce,
```

Comment:
So if we wanted to use this approach for defining extra resources related to a operator deployment. We wouldn't be able to template certain fields like [here](https://github.com/solo-io/gloo-mesh-enterprise/blob/7b2c7d5171ac9d41b36fc6cc6a2a8dcc167c804a/install/helm/gloo-platform/templates/redis-persistence.yaml#L15-L16). But then I asked myself do we really need to be allowing these fields to be configurable in the first place.

CC: @josh-pritchard @ilackarms